src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6846,7 +6846,7 @@ private BoundExpression BindLeftIdentifierOfPotentialColorColorMemberAccess(Iden
                                 // NOTE: ReplaceTypeOrValueReceiver will call CheckValue explicitly.
                                 boundValue = BindToNaturalType(boundValue, valueDiagnostics);
                                 return new BoundTypeOrValueExpression(left,
-                                    new BoundTypeOrValueData(leftSymbol, boundValue, valueDiagnostics.ToReadOnlyAndFree(), boundType, typeDiagnostics.ToReadOnlyAndFree()), leftType);
```

Comment:
>ToReadOnlyAndFree [](http://example.com/codeflow?start=102&length=17)

I think we need to review other usages of this API and evaluate if they could occur during attribute binding (even if only in an error scenario). Might want to adjust them as well.

Reply:
Perhaps we should never resolve diagnostics while creating instances of ```ImmutableBindingDiagnostic```

Reply:
Addressed.